rules/order/src/Rector/Class_/OrderPrivateMethodsByUseRector.php

```diff
@@ -86,32 +86,53 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
-        $desiredClassMethodOrder = $this->getLocalMethodCallOrder($node);
-        $privateClassMethodsByKey = $this->resolvePrivateClassMethods($node);
+        [$desiredPrivateMethodCallOrder, $privateClassMethodsByKey] = $this->getPrivateMethodCallOrderAndClassMethods(
+            $node
+        );
 
         // order is correct, nothing to change
-        if ($privateClassMethodsByKey === $desiredClassMethodOrder) {
+        if ($privateClassMethodsByKey === $desiredPrivateMethodCallOrder) {
             return null;
         }
 
         // different private method count, one of them is dead probably
-        if (count($desiredClassMethodOrder) !== count($desiredClassMethodOrder)) {
+        if (count($desiredPrivateMethodCallOrder) !== count($privateClassMethodsByKey)) {
             return null;
         }
 
-        $oldToNewKeys = $this->stmtOrder->createOldToNewKeys($desiredClassMethodOrder, $privateClassMethodsByKey);
+        $attempt = 0;
+        while (array_values($desiredPrivateMethodCallOrder) !== array_values($privateClassMethodsByKey)) {
+            $attempt++;
+            if ($attempt >= 5) {
```

Comment:
@TomasVotruba do you think we need a setting here? Or should it be enough to use constant?

Reply:
> Or should it be enough to use constant?

:+1:  Just so it's obvious there is something configurable in the class